locksmith/src/controllers/hookController.ts

```diff
@@ -0,0 +1,42 @@
+// eslint-disable-next-line import/no-unresolved
+import { Response } from 'express-serve-static-core'
+import { networks } from '@unlock-protocol/networks'
+import { SignedRequest } from '../types'
+import { Hook } from '../models'
+
+export function createHookHandler(topic: string) {
+  return async (req: SignedRequest, res: Response): Promise<any> => {
+    const { network, lock } = req.params
+    if (!networks[network]) {
+      return res.status(404).send({
+        error: `Network ${network} not supported!`,
+      })
+    }
+
+    const { callback, lease_seconds, secret, mode } = req.body
+
+    const [hook, created] = await Hook.findOrCreate<Hook>({
+      where: {
+        topic,
+        network,
+        lock,
+        callback: req.body.callback,
+      },
+      defaults: {
+        callback,
+        mode: 'subscribe',
+        topic,
+        network,
+        secret,
+        expiration: lease_seconds,
+        lock,
+      },
```

Comment:
I think we're missing the dialback part of the websub spec? It is important to add it in order to avoid spam...

Reply:
Yeah, I was going to add it.